### DISCUSSION TITLE
Add an option to output Zod schema with generated payload-types

### DISCUSSION THREAD

**Discussion by hades200082** [payloadcms/payload#4259] [Feature Requests & Ideas]
When generating payload-types.ts it would be nice if we could have the option to also output the associated Zod schemas as well. 

I've tried using ts-to-zod and @runtyping/zod to generate zod schemas but keep running into invalid union errors. :(

**Comment by smoove** [comment]
This would be super useful for using payload in conjunction with trpc procedure input parameters.

Api parameters and database rows are often pretty tightly coupled.
Assuming `PostCollectionSchema` is the zod schema for our post collection, we could very simply do this:

```ts
publicProcedure
    .input(PostCollectionSchema.pick({ title: true, body:true }))
    .mutation(async ({ ctx, input }) => {
      // input.title
      // input.body
})
```

**Reply by adriangonzy** [reply]
This would be awesome

**Comment by xndyz** [comment]
This would be a take Payload to another level, super helpful with forms too
